### ISSUE TITLE
https://vivo.sx

### ISSUE THREAD

**Issue by olegkre** [ytdl-org/youtube-dl#22328]
## Checklist

- [x ] I'm reporting a broken site support
- [x ] I've verified that I'm running youtube-dl version **2019.09.01**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
~# youtube-dl --get-url https://vivo.sx/9aed3c6992?#k08ep2pe -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--get-url', u'https://vivo.sx/9aed3c6992?#k08ep2pe', u'-v']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.09.01
[debug] Python version 2.7.6 (CPython) - Linux-3.13.0-100-generic-x86_64-with-Ubuntu-14.04-trusty
[debug] exe versions: avconv 9.20-6, avprobe 9.20-6, phantomjs 2.1.1
[debug] Proxy map: {}
ERROR: Unable to download webpage: <urlopen error [Errno 1] _ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure> (caused by URLError(SSLError(1, '_ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure'),))
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 627, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/YoutubeDL.py", line 2229, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/utils.py", line 2724, in https_open
    req, **kwargs)
  File "/usr/lib/python2.7/urllib2.py", line 1184, in do_open
    raise URLError(err)



# python3 `which youtube-dl` --no-check-certificate --get-url https://vivo.sx/9aed3c6992 -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-check-certificate', '--get-url', 'https://vivo.sx/9aed3c6992', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.12.09
[debug] Python version 3.4.3 (CPython) - Linux-3.13.0-100-generic-x86_64-with-Ubuntu-14.04-trusty
[debug] exe versions: avconv 9.20-6, avprobe 9.20-6, phantomjs 2.1.1
[debug] Proxy map: {}
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/__init__.py", line 472, in main
    _real_main(argv)
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/__init__.py", line 462, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/YoutubeDL.py", line 2002, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/YoutubeDL.py", line 793, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/extractor/common.py", line 508, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/extractor/shared.py", line 23, in _real_extract
    video_url = self._extract_video_url(webpage, video_id, url)
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/extractor/shared.py", line 95, in _extract_video_url
    'stream url', default=None, group='url')))
  File "/usr/local/lib/python3.4/dist-packages/youtube_dl/extractor/shared.py", line 91, in decode_url
    return compat_b64decode(encoded_url).decode('utf-8')
  File "/usr/lib/python3.4/base64.py", line 83, in b64decode
    s = _bytes_from_decode_data(s)
  File "/usr/lib/python3.4/base64.py", line 46, in _bytes_from_decode_data
    "string, not %r" % s.__class__.__name__) from None
TypeError: argument should be a bytes-like object or ASCII string, not 'NoneType'
```

## Description

No link is coming

**Comment by xwitty** [comment]
same with me
### Verbose log
```
$ youtube-dl -v https://vivo.sx/9aed3c6992
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://vivo.sx/9aed3c6992']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.09.01
[debug] Python version 2.7.9 (CPython) - Linux-4.19.46+-armv6l-with-debian-8.0
[debug] exe versions: avconv present, avprobe present, ffmpeg 4.1.3-static, ffprobe N-82759-g1f5630a, rtmpdump 2.4
[debug] Proxy map: {}
[Vivo] 9aed3c6992: Downloading webpage
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 474, in main
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 464, in _real_main
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2010, in download
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 26, in _real_extract
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 120, in _extract_video_url
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 116, in decode_url
  File "/usr/bin/youtube-dl/youtube_dl/compat.py", line 2947, in compat_b64decode
  File "/usr/lib/python2.7/base64.py", line 73, in b64decode
    return binascii.a2b_base64(s)
TypeError: must be string or buffer, not None
```

```
$ python3 `which youtube-dl` --no-check-certificate --get-url https://vivo.sx/9aed3c6992 -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-check-certificate', '--get-url', 'https://vivo.sx/9aed3c6992', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.09.01
[debug] Python version 3.4.2 (CPython) - Linux-4.19.46+-armv6l-with-debian-8.0
[debug] exe versions: avconv present, avprobe present, ffmpeg 4.1.3-static, ffprobe N-82759-g1f5630a, rtmpdump 2.4
[debug] Proxy map: {}
Traceback (most recent call last):
  File "/usr/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 474, in main
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 464, in _real_main
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2010, in download
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 26, in _real_extract
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 120, in _extract_video_url
  File "/usr/bin/youtube-dl/youtube_dl/extractor/shared.py", line 116, in decode_url
  File "/usr/lib/python3.4/base64.py", line 83, in b64decode
    s = _bytes_from_decode_data(s)
  File "/usr/lib/python3.4/base64.py", line 46, in _bytes_from_decode_data
    "string, not %r" % s.__class__.__name__) from None
TypeError: argument should be a bytes-like object or ASCII string, not 'NoneType'
```

**Comment by waahhhh** [comment]
Please check for duplicated issues. For example duplicated to: https://github.com/ytdl-org/youtube-dl/issues/21175.  

Fix is available in: https://github.com/ytdl-org/youtube-dl/pull/22279.

**Comment by byteholding** [comment]
Hey,

the updated the code again.
could someone please solve the issue again?
thanks

**Comment by CorgJR** [comment]
> Please check for duplicated issues. For example duplicated to: #21175.
> 
> Fix is available in: #22279.

The fix worked for a few weeks, now unfortunately its not working again. 

`computer@computer:~/Desktop/youtube-dl-vivo$ python -m youtube_dl https://vivo.sx/c1cb97082d#k2rfppm2
[Vivo] c1cb97082d: Downloading webpage
WARNING: "url" field is missing or empty - skipping format, there is an error in extractor
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/computer/Desktop/youtube-dl-vivo/youtube_dl/__main__.py", line 19, in <module>
    youtube_dl.main()
  File "youtube_dl/__init__.py", line 474, in main
    _real_main(argv)
  File "youtube_dl/__init__.py", line 464, in _real_main
    retcode = ydl.download(all_urls)
  File "youtube_dl/YoutubeDL.py", line 2018, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "youtube_dl/YoutubeDL.py", line 807, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "youtube_dl/YoutubeDL.py", line 862, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "youtube_dl/YoutubeDL.py", line 1588, in process_video_result
    if formats[0] is not info_dict:
IndexError: list index out of range
`

**Comment by CorgJR** [comment]
> Please check for duplicated issues. For example duplicated to: #21175.
> 
> Fix is available in: #22279.

Unfortunately it stopped working yesterday, was using for fix for awhile. They've changed it again.
